prboom2/src/s_sound.c

```diff
@@ -930,9 +932,11 @@ static int S_getChannel(void *origin, sfxinfo_t *sfxinfo, sfx_params_t *params)
     return channel_not_found;
 
   // Only allow one sound per origin
+  // Preserve the secret revealed sound, unless a new one is called
   for (cnum = 0; cnum < numChannels; cnum++)
     if (channels[cnum].active && channels[cnum].origin == origin &&
-        (comp[comp_sound] || channels[cnum].sfx_class == params->sfx_class))
+        (comp[comp_sound] || channels[cnum].sfx_class == params->sfx_class) &&
```

Comment:
Couldn't you just turn this into a `>=` comparison, so that sounds with a class "higher" in the enum are never interrupted by sounds "lower" in the hirarchy?

Reply:
Thats not what the previous code did.

Vanilla: Only 1 class allowed for the origin
Boom+: It allows both a sfx_class_none and a sfx_class_important for the same origin.

This fix adds a new sfx_class_secret allowed for that origin, and is not affected by the vanilla limit of 1.